bin/convert-owl.py

```diff
@@ -0,0 +1,113 @@
+"""Convert OWL to FHIR"""
+import os
+from argparse import ArgumentParser
+from typing import Dict
+
+import curies
+import requests
+from bioontologies import robot
+from linkml_runtime.loaders import json_loader
+from oaklib.converters.obo_graph_to_fhir_converter import OboGraphToFHIRConverter
+from oaklib.datamodels.obograph import GraphDocument
+from oaklib.interfaces.basic_ontology_interface import get_default_prefix_map
+
+
+# Vars
+BIN_DIR = os.path.dirname(os.path.realpath(__file__))
+PROJECT_DIR = os.path.join(BIN_DIR, '..')
+OUTDIR = os.path.join(PROJECT_DIR, 'output')
+
+
+# Functions
+def download(url: str, path: str):
+    """Download file at url to local path"""
+    _dir = os.path.dirname(path)
+    if not os.path.exists(_dir):
+        os.mkdir(_dir)
+    if not os.path.exists(path):
+        with open(path, 'wb') as f:
+            response = requests.get(url, verify=False)
+            f.write(response.content)
+
+
+def owl_to_obograph(inpath: str):
+    """Convert OWL to Obograph"""
+    parse_results: robot.ParseResults = robot.convert_to_obograph_local(inpath)
+    graph = parse_results.graph_document.graphs[0]
+    return graph
+
+
+# TODO: When https://github.com/INCATools/ontology-access-kit/pull/374 is merged, can add include_all_predicates
+def obograph_to_fhir(inpath: str, out_dir: str, out_filename: str = None, include_all_predicates=False):
+    """Convert Obograph to FHIR"""
+    converter = OboGraphToFHIRConverter()
+    converter.curie_converter = curies.Converter.from_prefix_map(get_default_prefix_map())
+    gd: GraphDocument = json_loader.load(inpath, target_class=GraphDocument)
+    out_path = os.path.join(out_dir, out_filename)
+    # converter.dump(gd, out_path, include_all_predicates=True)
+    converter.dump(gd, out_path)
+
+
+def owl_to_fhir(
+    input_path_or_url: str, out_dir: str = OUTDIR, out_filename: str = None, predicates_is_a_only=False,
+    retain_intermediaries=False
+):
+    """Run conversion"""
+    # Download if necessary
+    # TODO: determine if is URL or path. if URL, download and set new input_path
+    url = ''
+    input_path = input_path_or_url if not url else 'todo'
+    if url:
+        download(url, input_path)
+
+    # Convert to Obograph
+    obograph = owl_to_obograph(input_path)
+
+    # TODO: save obograph file
+    obograph_path = ''
```

Comment:
Still needs to be done:
> `    # TODO: save obograph file`

I forgot I would need to do this step. When I was writing my own converter, I just kept the Obograph in memory without saving to disk. But for the current OAK implementation, I need to save it and re-read.

Reply:
done